openfeature/hook/__init__.py

```diff
@@ -27,8 +26,9 @@ class HookContext:
 
 
 class Hook:
-    @abstractmethod
-    def before(self, hook_context: HookContext, hints: dict) -> EvaluationContext:
+    def before(
+        self, hook_context: HookContext, hints: dict
+    ) -> typing.Optional[EvaluationContext]:
```

Comment:
Could you please add a test that adds evaluation context in the before hook?

Reply:
Just added a unit test for full context evaluation merging which was missing.